rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -71,6 +73,10 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
   std::string get_storage_identifier() const override;
 
   uint64_t get_minimum_split_file_size() const override;
+  // APIs to handle/control transactions
+  void activate_transaction();
+
+  void commit_transaction();
```

Comment:
Why do these two functions have to be in the public interface? Ideally, the transactions should be opaque to the caller and should be handled solely by the sqlite plugin.